### PR TITLE
Module move and rename: spring4 -> spring

### LINKED ISSUE #2386: Could we move to next Spring?
There is the _feign-spring4_ module (`spring4`).

Could we deprecate this module and make new _feign-spring5_ or just rename module and artifact?

/cc @velo @kdavisk6

### PR DESCRIPTION
**Pull request by vitalijr2**
Resolve #2386 and some Spring issues.

### CODE REVIEW

**Comment by velo** [comment]
It would be nice to also include the relocation information, at least for the next release

**Comment by velo** [comment]
Ow, I was thinking about this
https://maven.apache.org/guides/mini/guide-relocation.html

In short, you leave behind a strip down pom for `fegin-spring4` and include `distributionManagement/relocation/artifactId`

**Comment by vitalijr2** [comment]
Interesting Maven feature, didn't know about it

File: spring4/pom.xml [hunk 1]
```diff
@@ -33,28 +33,11 @@
...
-      <artifactId>feign-jackson</artifactId>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
-
+  <distributionManagement>
+    <relocation>
+      <groupId>io.github.openfeign</groupId>
+      <artifactId>feign-spring</artifactId>
+      <version>13.3-SNAPSHOT</version>
```
Comment (velo) [inline spring4/pom.xml:40]: You can remove the version

**Review by velo — APPROVED** [review]

**Review by velo — APPROVED** [review]

**Pull request merged**